src/asset-swapper/quote_consumers/feature_rules/transform_erc20_rule.ts

```diff
@@ -85,6 +92,24 @@ export class TransformERC20Rule extends AbstractFeatureRule {
             });
         }
 
+        // Create an AffiliateFeeTransformer if there are fees in sell token.
+        // Must be before the FillQuoteTransformer.
+        if (sellTokenAffiliateFees.length > 0) {
+            transformations.push({
+                deploymentNonce: this.transformerNonces.affiliateFeeTransformer,
+                data: encodeAffiliateFeeTransformerData({
+                    fees: sellTokenAffiliateFees
+                        .filter((fee) => fee.sellTokenFeeAmount.gt(0))
+                        .map((fee) => ({
+                            token: sellToken, // TODO: handle native token
```

Comment:
For the use case for tx relay, it does not need to handle native token. However, I get this can potentially be used by flows other than tx relay